app/controllers/imports_controller.rb

```diff
@@ -0,0 +1,117 @@
+class ImportsController < ApplicationController
+
+  before_filter :admin_only
+
+  def import_school_district_json
+    file_data = params[:import][:import].read 
+    begin
+      json_data = JSON.parse file_data, :symbolize_names => true
+      if json_data[:districts].nil? || json_data[:schools].nil?
+        raise "Invalid JSON"
+      end
+    rescue => e
+      redirect_to import_school_district_status_imports_path({:message => "Invalid JSON"})
+      return
+    end
+    name = "upload_#{UUIDTools::UUID.timestamp_create.hexdigest}.json"
+    directory = "public/json"
+    path = File.join(directory, name)
+    dir = File.dirname(path)
+    FileUtils.mkdir_p(dir) unless File.directory?(dir)
+    File.open(path, "w") do |f|
+      f.write(file_data)
+    end    
+    import = Import.create!()
+    job = Delayed::Job.enqueue ImportSchoolsAndDistricts.new(import, path)
```

Comment:
This won't work in multiple server setup. Some of our portals are deployed that way.
Delayed Job stores the job in the database and the first worker that is free will take the job out of the database and start working on it. Since the file is stored in the file system of the server, if the job runs on a different server then it won't find the file. 

The easiest approach would be to put the file in the database. I don't think we'll hit any database size limits.